lib/simulator-xcode-6.js

```diff
@@ -184,18 +187,33 @@ class SimulatorXcode6 {
     return _.compact(existence).length !== files.length;
   }
 
-  //
-  // setPreferences () {
-  //
-  // }
-  //
-  // setLocationPreferences () {
-  //
-  // }
-  //
-  // setSafariPreferences () {
-  //
-  // }
+  async run () {
+    const STARTUP_TIMEOUT = 3000;
+    const BOOTING_TIMOUT = 20000;
+    const BOOTED_POLLING_INTERVAL = 200;
+
+    // start simulator
+    let simulatorApp = path.resolve(await getXcodePath(), 'Applications', 'Simulator.app');
+    let args = [simulatorApp, '--args', '-CurrentDeviceUDID', this.udid];
+    log.info(`starting simulator with command: open ${args.join(' ')}`);
+    await exec('open', args, {timeout: STARTUP_TIMEOUT});
+
+    //wait for it to be 'booted'
+    let checkBooted = async function () {
+      let stat = await this.stat();
+      return new B((resolve, reject) => {
```

Comment:
what is the point of creating a promise here? this whole function can go away and below you can just have as the anonymous async function:

```js
async () => {
  let stat = await this.stat();
  if (stat.state !== 'Booted') {
    throw new Error("State wasn't booted");
  }
}
```